src/Quaternion.jl

```diff
@@ -135,50 +133,31 @@ Compute the quaternion conjugate of a quaternion `q`.
 # Examples
 ```jldoctest
 julia> conj(Quaternion(1,2,3,4))
-Quaternion{Int64}(1, -2, -3, -4, false)
+Quaternion{Int64}(1, -2, -3, -4)
 ```
 """
-Base.conj(q::Quaternion) = Quaternion(q.s, -q.v1, -q.v2, -q.v3, q.norm)
+Base.conj(q::Quaternion) = Quaternion(q.s, -q.v1, -q.v2, -q.v3)
 Base.abs(q::Quaternion) = sqrt(abs2(q))
 Base.float(q::Quaternion{T}) where T = convert(Quaternion{float(T)}, q)
 abs_imag(q::Quaternion) = sqrt(q.v2 * q.v2 + (q.v1 * q.v1 + q.v3 * q.v3)) # ordered to match abs2
 Base.abs2(q::Quaternion) = (q.s * q.s + q.v2 * q.v2) + (q.v1 * q.v1 + q.v3 * q.v3)
-Base.inv(q::Quaternion) = q.norm ? conj(q) : conj(q) / abs2(q)
+Base.inv(q::Quaternion) = conj(q) / abs2(q)
 
 Base.isreal(q::Quaternion) = iszero(q.v1) & iszero(q.v2) & iszero(q.v3)
-Base.isfinite(q::Quaternion) = q.norm | (isfinite(q.s) & isfinite(q.v1) & isfinite(q.v2) & isfinite(q.v3))
-Base.iszero(q::Quaternion) = ~q.norm & iszero(real(q)) & iszero(q.v1) & iszero(q.v2) & iszero(q.v3)
+Base.isfinite(q::Quaternion) = isfinite(q.s) & isfinite(q.v1) & isfinite(q.v2) & isfinite(q.v3)
+Base.iszero(q::Quaternion) = iszero(real(q)) & iszero(q.v1) & iszero(q.v2) & iszero(q.v3)
 Base.isnan(q::Quaternion) = isnan(real(q)) | isnan(q.v1) | isnan(q.v2) | isnan(q.v3)
-Base.isinf(q::Quaternion) = ~q.norm & (isinf(q.s) | isinf(q.v1) | isinf(q.v2) | isinf(q.v3))
+Base.isinf(q::Quaternion) = isinf(q.s) | isinf(q.v1) | isinf(q.v2) | isinf(q.v3)
 
-function LinearAlgebra.normalize(q::Quaternion)
-    if (q.norm)
-        return q
-    end
-    q = q / abs(q)
-    Quaternion(q.s, q.v1, q.v2, q.v3, true)
-end
-
-function normalizea(q::Quaternion)
-    if (q.norm)
-        return (q, one(q.s))
-    end
-    a = abs(q)
-    q = q / a
-    (Quaternion(q.s, q.v1, q.v2, q.v3, true), a)
-end
+# included strictly for documentation; the base implementation is sufficient
+"""
+    sign(q::Quaternion) -> Quaternion
 
-function normalizeq(q::Quaternion)
-    a = abs(q)
-    if a > 0
-        q = q / a
-        Quaternion(q.s, q.v1, q.v2, q.v3, true)
-    else
-        Quaternion(0.0, 1.0, 0.0, 0.0, true)
-    end
-end
+Return zero if `q==0` and `q/|q|` otherwise.
```

Comment:
I think using Julia format
```suggestion
Return zero if `q==0` and `q/abs(q)` otherwise.
```
or math expression mode
```suggestion
Return zero if ``q=0`` and ``q/|q|`` otherwise.
```
would be better. If we follow the docstring of `Base.sign`, it will be like this:
```suggestion
Return zero if `q==0` and ``q/|q|`` otherwise.

# Examples
```jldoctest
julia> sign(Quaternion(4,0,0,0))
QuaternionF64(1.0, 0.0, 0.0, 0.0)

julia> sign(Quaternion(1,1,0,0))
QuaternionF64(0.7071067811865475, 0.7071067811865475, 0.0, 0.0)
```